### ISSUE TITLE
Documentation issues

### ISSUE THREAD

**Issue by dsamx** [openSUSE/obs-docu#74]
I found several issues that I want to track in this thread.
If I knew OBS I could help with the documentation but these gaps are just too big for me.

The books have no real reading order but they depend on each other; in package terms they have several circular dependencies on each other, depending on the topic of interest.
There should either be a clear reading order or every book has to ensure that required information was printed before it is mentioned somewhere.
Examples will follow.

= Best practices:
2.1 only contains the text "Binaries are the".
-> "These books are considered to contain only reviewed content, establishing the reference documentation of OBS." yeah, sorry, but no.

It is not explained/linked how package tracking works (6.2).
6.3 does not explain what the parameters really do; the obs application for example does not have a /var/tmp/build-root directory. Which permissions does the directory need, which is the default if the documented default does not exist?
API communication: It should be mentioned that on a private OBS installation the apiurl entry in .oscrc should be changed to point to the private server instead of api.opensuse.org because it is unclear to new users what that entry is for.

Chapter 7 and 8 are empty

Text explaining how to create a project should link to the project metadata section in the Ref guide

= Reference guide
3.2 The API is empty
3.5.1 mentions that git, svn and cvs are SCMs but does not describe how to configure them. There is only a shortened git example of a _service file (which is used how?). It should also be explained how branches and tags of a package in an SCM can be used or if this is not possible. Only building the HEAD does not seem to be useful for releases.
3.6, 3.7 4.1 are empty and important
4.2.1 does not mention or link to how to configure these options
4.2.2: typo: "maschine"
4.3, 4.4 empty and important
Chapter 5 is sparse, 5.1.7 and .8 are empty
The introduction to Chapter 6 is empty
Chapter 7 misses an example through the request&review system. Maintenance releases and package tracking are important and not further described here. There should be a link to chapter 15.
8.2 is likely incomplete. ("Each format has some specials")
8.2.5 is empty
11.5 and .6 are empty
The chapters 13 and 14 are completely sparse.
It is unclear how to configure a link between a package and bugzilla. Is this done in the _patchinfo file? How exactly?
Chapter 17 is too sparse to understand how this works or has to be configured.
18.3.2 is sparse, 18.5 and 18.6 are empty
19.2 is cut off in the pdf version of the documentation
23 does not explain how to disable all or selected checks
24 is incomplete
Appendix A is empty

There is no information about how to import proprietary packages which are build requirements. Assume to require a closed source rpm containing a shared library or oracle jdk or something like this for a package of yours.
There is no (link to a) documentation about how to properly backup an OBS server system.

= admin guide
References to the OBS server are inconsistent: best practices e.g. uses https://vm.ip.address and the admin guide uses https://api.testobs.org. The latter also implies that the api vhost may exist or may have to exist. It should be noted before any other osc commands that configuration of the local apiurl is mandatory; otherwise it defaults to api.opensuse.org where the user is unlikely to have any permissions to check in his configuration - or the example commands have to use the -A parameter all the time.

The guide should mention where hostname changes should be applied to: system, apache, obs, wherever.

2.3.1, 2.3.6, 2.3.13 just prints some ugly hex stuff in the pdf:
```
&#x251c;&#x2500;&#x2500; openSUSE:13.2
&#x2502;~~ &#x2514;&#x2500;&#x2500; standard
[...]
```

2.3 is sparse; e.g. "2.3.4 events directory": Is this directory important? How does the mentioned communication work? Is something required to do, is it just nice for debugging purposes? Or can this be left out of the documentation because it is some OBS internal stuff no admin should be caring about?

2.4.3 typo: setiings
2.4.4 typo: atrribute

Table 2.7 and probably others do not explain the possible values; e.g. publish is for optional publish flags. Which ones?

3.1 just includes the output of obs_admin --help without further explanation. As the last checkin of the documentation was several months ago, this information is likely deprecated.
3.1.2 "for some taks, admin people also need this tool". I immediately wanted to know which ones but I had to skipf several sentences and examples and a warning block to find that information. That should be rephrased for better readability

3.2 The interconnect did not work with the OBS application VM after some serious xml copy-pasting and trying this and that on the command line. This part has to be tested and documented with the current release.
Mirroring with obs_mirror_project worked like that:
osc -A https://api.opensuse.org api /distributions > distributions.xml
edit and upload locally with
osc api /distributions -T distributions.xml
Then
wget https://raw.githubusercontent.com/openSUSE/open-build-service/master/dist/obs_mirror_project
zypper install ruby
Then create new empty projects for each distribution and add some advanced/meta xml like that:

```
<project name="SUSE:SLE-11:SP4">
  <title>SLES11 SP4 build repositories</title>
  <description>SLES11 SP4 build repositories</description>
  <person userid="Admin" role="maintainer"/>
  <build>
    <disable/>
  </build>
  <publish>
    <disable/>
  </publish>
  <repository name="standard">
    <arch>x86_64</arch>
  </repository>
</project>
```

Install the build packages like that
./obs_mirror_project -p SUSE:SLE-11:SP4 -r standard -a x86_64 -d /opt/distributions
Note: do not install into /srv/obs/build because this screws with the configuration of the webui and vice versa. Only copy the rpm files to /srv/obs/build/name/standard/x86_64/:full and ensure everything belongs to obsrun:obsrun.
Then rescan with obs_admin --rescan-repository SUSE:SLE-11:SP4 standard x86_64 or whichever name was chosen and control for errors in /srv/obs/log/scheduler_x86_64.log
The current 3.2.2.1 does not cover that in the slightest.

3.2 does not mention how to create own rpm lists, e.g. for RHEL which has some nice licensing issues so there are no packages available. I used obs_mirror_project to download the package list and had to download and rename (as in remove all version information from the filename) the packages from RedHat.

= Other sources
Information from
- https://en.opensuse.org/openSUSE:Build_Service_Tips_and_Trick
- https://en.opensuse.org/openSUSE:Build_Service_cross_distribution_howto
- https://en.opensuse.org/openSUSE:Build_Service_prjconf
- and who knows where
should be included into the books.

**Comment by tomschr** [comment]
> The books have no real reading order but they depend on each other; in package terms they have several circular dependencies on each other, depending on the topic of interest.

I agree. However, Stefan and I try to tackle this problem. We worked in the last weeks on the structure and tried to improve it. It is by no means perfect, but IMHO it's a lays the foundation for further improvements.

In master you have now:

* A new [Beginnerʼs Guide](http://openbuildservice.org/help/manuals/obs-beginners-guide/)
  This guide is intended for users and developers interested in building packages and gives a short overview of the typical workflows with OBS. We consider the guide mostly as finished. Of course, feedback is always appreciated.

* A new [User Guide](http://openbuildservice.org/help/manuals/obs-user-guide/)
  This guide is intended for users and developers who want to dig deeper into OBS.  It contains more background information, setup, and usage scenarios. This guide is work in progress.

* A revised [Reference Guide](http://openbuildservice.org/help/manuals/obs-reference-guide/)
  This guide is intended for the administrator of an OBS server. Some chapters or sections which were more useful for a user than an administrator were moved into the User Guide. This guide is unfinished.

* [Best Practices Guide](http://openbuildservice.org/help/manuals/obs-best-practices/) & [Administration Guide](http://openbuildservice.org/help/manuals/obs-admin-guide/)
  Mostly unchanged except for some title and other minor XML corrections. Mostly untouched.

@dsamx
I will go through your issues in more detail later, but this is just a first answer. 😉

**Comment by dsamx** [comment]
thanks :-) Could you focus on how to configure a distribution? I somehow made it work in a previous attempt but now I made a fresh obs installation and thought my notes from my first try would suffice but I still don't get how to properly setup a distribution with obs_mirror_project.
